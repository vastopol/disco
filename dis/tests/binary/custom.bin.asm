ILLEGAL_OPCODE

RTI

TRAP x20
TRAP x21
TRAP x22
TRAP x23
TRAP x24
TRAP x25

JMP R0
JMP R1
JMP R2
JMP R3
JMP R4
JMP R5
JMP R6
JMP R7

BR x0
BRp x0
BRz x0
BRzp x0
BRn x0
BRnp x0
BRnz x0
BRnzp x0

LD R0, x0
LD R1, x0
LD R2, x0
LD R3, x0
LD R4, x0
LD R5, x0
LD R6, x0
LD R7, x0

LDI R0, x0
LDI R1, x0
LDI R2, x0
LDI R3, x0
LDI R4, x0
LDI R5, x0
LDI R6, x0
LDI R7, x0

LEA R0, x0
LEA R1, x0
LEA R2, x0
LEA R3, x0
LEA R4, x0
LEA R5, x0
LEA R6, x0
LEA R7, x0

ST R0, x0
ST R1, x0
ST R2, x0
ST R3, x0
ST R4, x0
ST R5, x0
ST R6, x0
ST R7, x0

STI R0, x0
STI R1, x0
STI R2, x0
STI R3, x0
STI R4, x0
STI R5, x0
STI R6, x0
STI R7, x0

NOT R0, R0
NOT R1, R1
NOT R2, R2
NOT R3, R3
NOT R4, R4
NOT R5, R5
NOT R6, R6
NOT R7, R7

LDR R0, R0, x0
LDR R1, R1, x1
LDR R2, R2, x2
LDR R3, R3, x3
LDR R4, R4, x4
LDR R5, R5, x5
LDR R5, R5, xD
LDR R7, R7, x3F

AND R0, R0, R0
AND R1, R1, R1
AND R2, R2, R2
AND R0, R0, x0
AND R1, R1, x1
AND R2, R2, x2

ADD R0, R0, R0
ADD R1, R1, R1
ADD R2, R2, R2
ADD R0, R0, x0
ADD R1, R1, x1
ADD R2, R2, x2

JSR x0
JSRR R0
