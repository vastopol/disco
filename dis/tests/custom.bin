1101000000000000

1000000000000000

1111000000100000
1111000000100001
1111000000100010
1111000000100011
1111000000100100
1111000000100101

1100000000000000
1100000001000000
1100000010000000
1100000011000000
1100000100000000
1100000101000000
1100000110000000
1100000111000000

0000000000000000
0000001000000000
0000010000000000
0000011000000000
0000100000000000
0000101000000000
0000110000000000
0000111000000000

0010000000000000
0010001000000000
0010010000000000
0010011000000000
0010100000000000
0010101000000000
0010110000000000
0010111000000000

1010000000000000
1010001000000000
1010010000000000
1010011000000000
1010100000000000
1010101000000000
1010110000000000
1010111000000000

1110000000000000
1110001000000000
1110010000000000
1110011000000000
1110100000000000
1110101000000000
1110110000000000
1110111000000000

0011000000000000
0011001000000000
0011010000000000
0011011000000000
0011100000000000
0011101000000000
0011110000000000
0011111000000000

1011000000000000
1011001000000000
1011010000000000
1011011000000000
1011100000000000
1011101000000000
1011110000000000
1011111000000000

1001000000111111
1001001001111111
1001010010111111
1001011011111111
1001100100111111
1001101101111111
1001110110111111
1001111111111111

0110000000000000
0110001001000001
0110010010000010
0110011011000011
0110100100000100
0110101101000101
0110101101001101
0110111111111111

