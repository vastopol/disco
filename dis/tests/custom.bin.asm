ILLEGAL_OPCODE

RTI

JMP R0
JMP R1
JMP R2
JMP R3
JMP R4
JMP R5
JMP R6
JMP R7

BR 000000000
BRp 000000000
BRz 000000000
BRzp 000000000
BRn 000000000
BRnp 000000000
BRnz 000000000
BRnzp 000000000
