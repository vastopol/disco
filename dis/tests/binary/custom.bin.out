D000

8000

F020
F021
F022
F023
F024
F025

C000
C040
C080
C0C0
C100
C140
C180
C1C0

0000
0200
0400
0600
0800
0A00
0C00
0E00

2000
2200
2400
2600
2800
2A00
2C00
2E00

A000
A200
A400
A600
A800
AA00
AC00
AE00

E000
E200
E400
E600
E800
EA00
EC00
EE00

3000
3200
3400
3600
3800
3A00
3C00
3E00

B000
B200
B400
B600
B800
BA00
BC00
BE00

903F
927F
94BF
96FF
993F
9B7F
9DBF
9FFF

6000
6241
6482
66C3
6904
6B45
6B4D
6FFF

5000
5241
5482
5020
5261
54A2

1000
1241
1482
1020
1261
14A2

4800
4000
